### ISSUE TITLE
Messages not being copied (deepcopy) due to regex match object

### ISSUE THREAD

**Issue by andreleblanc11** [MetPX/sarracenia#857]
Encountered on development.

Error message generated.
```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/sarracenia/flow/__init__.py", line 1093, in post
    p(self.worklist)
  File "/usr/lib/python3/dist-packages/sarracenia/flowcb/post/message.py", line 43, in post
    if all_good and hasattr(self.poster,'putNewMessage') and self.poster.putNewMessage(m):
  File "/usr/lib/python3/dist-packages/sarracenia/moth/amqp.py", line 625, in putNewMessage
    body = copy.deepcopy(message)
  File "/usr/lib/python3.6/copy.py", line 180, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/usr/lib/python3.6/copy.py", line 306, in _reconstruct
    value = deepcopy(value, memo)
  File "/usr/lib/python3.6/copy.py", line 161, in deepcopy
    y = copier(memo)
TypeError: cannot deepcopy this match object
```

After the error, sarracenia still indicates in the logs that a message is posted, even though, looking downstream, no message is to be found.

```
2023-12-04 18:41:29,378 [INFO] sarracenia.flowcb.log after_post posted to a directory with baseUrl: ftp://$USER@$SERVER relPath: data/depot
/pds/users/airnow2/SK
2023-12-04 18:41:29,378 [INFO] sarracenia.flowcb.log after_post posted to a directory with baseUrl: ftp://$USER@$SERVER relPath: data/depot
/pds/users/airnow2/TEST
```

https://stackoverflow.com/questions/6279305/typeerror-cannot-deepcopy-this-pattern-object/52885510#52885510 suggests that the issue stems from an incompatibility for `deepcopy` to interpret `re` objects when versions lower then Python3.7 are used. This seems to have been validated. The same configuration was tested with Python3.10 and doesn't have this same problem.

There is a proposed work around.. https://stackoverflow.com/a/56935186

`_matches` looks to be the option being `re` object

```
'_matches':  <re.Match object; span=(0, 64), match='ftp://$USER@$SERVER/$DIRECTORY>
```

https://github.com/MetPX/sarracenia/blob/development/sarracenia/flow/__init__.py#L1010 looks to be where `_matches` is being set in the message.

**Comment by petersilva** [comment]
* on redhat8 and ubuntu 18, python3.6 is used, so need a work-around for older OS releases.
* github stopped letting us run flow tests on ubuntu 18.04, so need to find a solution to run 18.04 tests.  We should have caught this prior to release.

**Comment by andreleblanc11** [comment]
It looks like the error might have been introduced in V3.00.48..

```
sarra@dev-server:~/.cache/sr3/log$ grep -m1 'cannot deepcopy' poll_$CONFIG_01.log.2023-12-01
2023-12-01 19:07:51,886 [ERROR] sarracenia.flow post flowCallback plugin <bound method Message.post of <sarracenia.flowcb.post.message.Message object at 0x7f92676e50f0>> crashed: cannot deepcopy this match object
sarra@dev-server:~/.cache/sr3/log$ head -n 5 /var/log/apt/history.log

Start-Date: 2023-12-01  19:03:20
Commandline: apt-get install metpx-sr3
Upgrade: metpx-sr3:amd64 (3.00.47~ubuntu18.04.1, 3.00.48~ubuntu18.04.1)
End-Date: 2023-12-01  19:03:21
```

**Comment by andreleblanc11** [comment]
It also affects a lot of the polls on that same dev server

**Comment by andreleblanc11** [comment]
This is the change that instantiated this error .. from the V3.00.48 commits https://github.com/MetPX/sarracenia/commit/aada6f0f4f4dfaa2bc467bd9a24b7566645604f4#diff-e651cf920899e45ce83b7ec6f89e2b69ae20c17bf5ee15a9f1b182550595c7ea

**Comment by petersilva** [comment]
yes, the change is the  main implementation of the fix for #839 which was the main point of v3.00.48.  installing 3.00.48 without that feature doesn't make much sense (might as well use 3.00.47.)  I guess downgrading to 47 is a good work-around.

**Comment by petersilva** [comment]
so... the dynamic_flow tests were failing for a while, and it turned out to be the result of:

* hpfx adding restricted download radar data, for which retrievals always fail.
* hpfx model_rdps downloads being corrupt, resulting in retrieval failures 
* the above were fixed by changes do product selection in sr_insects 

* apparently, some messages are being published with malformed dates, and sr3 isn't handling that well, resulted in two problems:
* 3fb9150d79e04984ad6b0756e139ca85a7277344 ... the acks were not getting sent when a malformed message was received.
* ab0f786debc5b264e5512b37b537fa1f97b69069 ... it just says "malformed message" want more.

With all of the above applied, the dynamic flow tests pass on 22.04 at least, and the initial fixes were tested by @andreleblanc11 on the internal system where it showed up in the first place.
